strangerstudios/paid-memberships-pro
# Feature/stripe modal on success

### All Submissions:

<img width="961" alt="image" src="https://github.com/strangerstudios/paid-memberships-pro/assets/1678457/11f4d666-5cdf-4c90-abe7-4f1f18068e8f">
<img width="605" alt="image" src="https://github.com/strangerstudios/paid-memberships-pro/assets/1678457/f43efd21-b64b-43e2-828b-ac94dfa12204">


<img width="1304" alt="Screenshot 2023-07-29 at 6 45 42 PM" src="https://github.com/strangerstudios/paid-memberships-pro/assets/1678457/46072495-2f59-4c75-a21a-eba4e70f725f">
<img width="1473" alt="Screenshot 2023-07-29 at 6 46 48 PM" src="https://github.com/strangerstudios/paid-memberships-pro/assets/1678457/1ffb2dd5-badf-4894-949e-ac023cddf1ba">


### Changes proposed in this Pull Request:

In Stripe dedicated class adds an static function on admin_init hook and there check if incoming request contains the `pmpro_stripe_publishable_key` parameter. If yes, adds to page DOM a modal (hidden) telling the user the connection to stripe was ok. Modal is shown later, on DOM ready from admin JS file with jQuery.


### How to test the changes in this Pull Request:

1. Go to payment gateway page.
2. If connected to stripe disconnect.
3.  Connect again
4. Wait until Stripe redirect to the payment gateway page again and observe if the modal show.

### Other information:



### Changelog entry

 [Enhancement] Stripe success callback page has now a modal telling the user is connected.
